operator/pkg/ingress/ingress.go

```diff
@@ -208,15 +208,8 @@ func (ic *IngressController) handleIngressUpdatedEvent(event ingressUpdatedEvent
 }
 
 func (ic *IngressController) handleIngressDeletedEvent(event ingressDeletedEvent) error {
-	// as ciliumEnvoyConfig is non-namespace scoped, it should be removed manually.
-	// other resources (e.g service, endpoints) will be removed automatically via ownerReferences
 	log.WithField(logfields.Ingress, event.ingress.Name).Debug("Deleting CiliumEnvoyConfig for ingress")
-
 	ic.secretManager.Add(event)
-	if err := ic.deleteCiliumEnvoyConfig(event.ingress); err != nil {
-		log.WithError(err).Warn("Failed to delete cilium-envoy-config")
-		return err
-	}
```

Comment:
I think this code should be kept for the same reason that we have added this https://github.com/cilium/cilium/pull/19576

Reply:
I don't quite understand, can you elaborate further ? 

AFAIK, this CEC is created and owned by Ingress object, so the cleanup should be offloaded to k8s via ownerReferences.

Reply:
discussed offline, this could be the case that CEC can become orphan at its creation if the Ingress is deleted at the same time the CEC is being created.

Reply:
Discussed offline